go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/argoproj/argo-rollouts
 go 1.16
 
 require (
+	github.com/PaesslerAG/jsonpath v0.1.1
```

Comment:
Hmmm, this library uses "BSD 3-Clause "New" or "Revised" License" which is not what we typically use (Apache 2 or MIT). I'm not sure this is okay to use. Is there something else we can use or write our own?

Reply:
I tried using a different library called `objx`, which was similar. But the problem there was that using a non-existing path (`path: "some.none.existing[4].path.in.rollout"`) just returns an empty string. So it's not possible to display an error when the user used an invalid path or not (unless we assume an empty string is always an error).

If you know of any other possible libraries, I'd be happy to give them a shot as well.
I can also write a simple parser by splitting on `"."`. I don't think it will be too complicated, given our very specific use-case (specifically with field names that do not contain dots themselves)